python/lsst/meas/extensions/convolved/convolved.py

```diff
@@ -0,0 +1,503 @@
+#
+# LSST Data Management System
+# Copyright 2008-2016 AURA/LSST.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+
+import math
+import numpy as np
+
+from lsst.pex.config import Field, ListField, ConfigField, makeConfigClass
+from lsst.pipe.base import Struct
+from lsst.meas.extensions.photometryKron import KronAperture, KronFluxPlugin
+from lsst.meas.base.flagDecorator import addFlagHandler
+
+import lsst.meas.base
+import lsst.afw.math
+import lsst.afw.geom
+import lsst.afw.image
+
+__all__ = ("ConvolvedFluxConfig", "ConvolvedFluxPlugin")
+
+
+SIGMA_TO_FWHM = 2.0*math.sqrt(2.0*(math.log(2.0)))  # Multiply sigma by this to get FWHM
+PLUGIN_NAME = "ext_convolved_ConvolvedFlux"  # Usual name for plugin
+
+
+class DeconvolutionError(RuntimeError):
+    """Convolving to the target seeing would require deconvolution"""
+    pass
+
+
+class NoKronError(RuntimeError):
+    """There's no Kron radius available"""
+    pass
+
+
+ApertureFluxConfig = makeConfigClass(lsst.meas.base.ApertureFluxControl)
```

Comment:
It's a full-fledged class, used within one of the other classes.  I think defining it at the top level should be preferred to defining it within the `ConvolvedFluxConfig` class.  It's not exported by default.